### ISSUE TITLE
DuckDB can't read from TimescaleDB hypertables

### ISSUE THREAD

**Issue by theelderbeever** [duckdb/postgres_scanner#130]
### What happens?

I am using the postgres extension TimescaleDB to handle an `events` table. Under the hood this is a timescaledb hypertable. When attaching with DuckDB's postgres scanner the scanner fails to read from the hypertable. I believe this has something to do with the compressed chunks on the hypertable. I am aware that this isn't vanilla postgres however, tools like `connectorx` and `polars` from the python/rust world is able to successfully read from the same table without issue.

```sql
D CALL postgres_attach('dbname=billing_platform user=postgres password=password host=127.0.0.1', source_schema='usage');
```
> Error: IO Error: Unable to query Postgres: ERROR:  transparent decompression only supports tableoid system column
 ERROR:  transparent decompression only supports tableoid system column

### To Reproduce

To start you can launch a TimescaleDB instance with the following compose file.

```yaml
# docker-compose.yaml
version: '3.7'
services:
  db:
    image: timescale/timescaledb-ha:pg15-ts2.12-all
    restart: always
    platform: linux/amd64
    environment:
      TZ: UTC
      POSTGRES_PASSWORD: password
      POSTGRES_DB: sandbox
    ports:
      - 5432:5432
```

Create a hypertable in timescale. You can connect with any postgres client such as psql
```sh
PGPASSWORD=password psql -h localhost -p 5432 -U postgres -d sandbox
```

```sql
-- TimescaleDB hypertable creation
CREATE TABLE IF NOT EXISTS events (
                timestamp TIMESTAMPTZ NOT NULL,
                eventname TEXT NOT NULL,
                account UUID NOT NULL,
                counter DOUBLE PRECISION
            );

            CREATE UNIQUE INDEX events_pk ON events (account, eventname, timestamp ASC);

            CREATE INDEX events_eventnametimestamp ON events (eventname, timestamp ASC);

            SELECT CREATE_HYPERTABLE('events', 'timestamp', chunk_time_interval => INTERVAL '15 minutes');

            ALTER TABLE events SET (
                timescaledb.compress,
                timescaledb.compress_orderby = 'timestamp ASC',
                timescaledb.compress_segmentby = 'account,eventname'
            );

            SELECT ADD_COMPRESSION_POLICY('events', INTERVAL '15 minuntes');
```

Insert some data
```sql
INSERT INTO events (timestamp, eventname, account, counter)
VALUES 
  (NOW(), 'some-event', '69735c6b-f764-4620-b476-c668f7c0ae48', 1.2);
```

Now in DuckDB run the following.
```sql
INSTALL postgres_scanner;
LOAD postgres_scanner;
CALL postgres_attach('dbname=billing_platform user=postgres password=password host=127.0.0.1', source_schema='public');
PRAGMA show_tables;
SELECT COUNT(*) FROM events;
```

If you fire all these at it quickly the problem may not occur since the table partitions won't have been compressed yet. They _should_ compress after about 15 minutes given the policy applied in the database setup.

### OS:

MacOS/docker

### PostgreSQL Version:

pg15.4/ts2.12

### DuckDB Version:

v0.9.1 401c8061c6

### DuckDB Client:

brew

### Full Name:

Taylor Beever

### Affiliation:

QuickNode

### Have you tried this on the latest `main` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by Mytherin** [comment]
Thanks for the report! Could you try this with v0.9.2? We reworked the extension quite significantly since 0.9.1.

**Comment by Mytherin** [comment]
I've managed to reproduce the issue - note that it is limited to `COUNT(*)` in the latest version. `SELECT * FROM events` works as expected. I've pushed a fix for this in #132.

**Comment by theelderbeever** [comment]
@Mytherin I guess I should have tested a few other queries... Thanks for the quick turnaround on this!
